db/migrate/20170724205314_add_secret_shares_to_moments.rb

```diff
@@ -0,0 +1,6 @@
+class AddSecretSharesToMoments < ActiveRecord::Migration
+  def change
+    add_column :moments, :secret_share_identifier, :string
+    add_column :moments, :secret_share_expires_at, :datetime
```

Comment:
Have you had a discussion about database indices?
When loading the share URL, you are going to need to find on the moments table for a secret_share_identifier. If there is no index on this field, the database is going to need to check every record looking for a match. In development, you'd never notice this but, when you have lots of records in production, this can be extraordinarily slow.